Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -123,7 +132,21 @@ buildEndRideHandle merchantId = do
       }
 
 driverEndRide ::
-  (MonadThrow m, Log m, MonadTime m, MonadGuid m, CoreMetrics m, MonadReader r m, HasField "enableAPILatencyLogging" r Bool, HasField "enableAPIPrometheusMetricLogging" r Bool) =>
+  ( MonadThrow m,
+    Log m,
+    MonadTime m,
+    MonadGuid m,
+    CoreMetrics m,
+    MonadReader r m,
+    HasField "enableAPILatencyLogging" r Bool,
+    HasField "enableAPIPrometheusMetricLogging" r Bool,
+    HasFlowEnv m r '["smsCfg" ::: SmsConfig],
+    HasCacheConfig r,
+    EsqDBFlow m r,
+    EsqLocDBFlow m r,
+    EncFlow m r,
+    Redis.HedisFlow m r
```

Comment:
All these are redundant, because included in `HasFlowEnv`
```
    MonadThrow m,
    Log m,
    MonadTime m,
    MonadGuid m,
    MonadReader r m,
```

Reply:
added the function in the servicehandle